File: Cap 04/Topico 4.5/Topico 4.5.3/Decisao por Selecao/decsel.scala
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 4 - Topico 4.5.3
 * 
 * Programa ..: OPERACAO COM DECISAO SELETIVA
 * 
 */

object Main {
  var MES: Int = 0
  def main(args: Array[String]): Unit = {
    println("MES POR EXTENSO")
    println()
    print("Entre o numero do mes: ")
    MES = scala.io.StdIn.readInt()
    MES match {
      case  1 => println("Janeiro")
      case  2 => println("Fevereiro")
      case  3 => println("Marco")
      case  4 => println("Abril")
      case  5 => println("Maio")
      case  6 => println("Junho")
      case  7 => println("Julho")
      case  8 => println("Agosto")
      case  9 => println("Setembro")
      case 10 => println("Outubro")
      case 11 => println("Novembro")
      case 12 => println("Dezembro")
      case  _ => println("Valor invalido")
    }
  }
}

File: Cap 04/Topico 4.5/Topico 4.5.3/Decisao por Selecao/decsel.lua
--[[
 * Livro .....: Algoritmos
 *              Logicaz para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 4 - Topico 4.5.3
 *
 * Programa ..: OPERACAO COM DECISAO SELETIVA (SIMULADA)
--]]

print("MES POR EXTENSO\n")
io.write("Entre o numero do mes: ")
MES = tonumber(io.read())
if (MES == 1) then
  print("Janeiro")
elseif (MES == 2) then
  print("Fevereiro")
elseif (MES == 3) then
  print("Marco")
elseif (MES == 4) then
  print("Abril")
elseif (MES == 5) then
  print("Maio")
elseif (MES == 6) then
  print("Junho")
elseif (MES == 7) then
  print("Julho")
elseif (MES == 8) then
  print("Agosto")
elseif (MES == 9) then
  print("Setembro")
elseif (MES == 10) then
  print("Outubro")
elseif (MES == 11) then
  print("Novembro")
elseif (MES == 12) then
  print("Dezembro")
else
  print("Valor invalido")
end
